foliage.cabal

```
@@ -38,32 +65,23 @@ executable foliage
         Foliage.Utils.GitHub
         Network.URI.Orphans
 
-    default-language:   GHC2021
+executable foliage
+    import:             shared-modules
+    main-is:            Main.hs
 
-    ghc-options:        -Wall
-    ghc-options:        -threaded
+library libfoliage
```

```suggestion
library
```
I think you can do this, executables and libraries don't occupy the same namespace.